cuda_core/cuda/core/experimental/__init__.py

```diff
@@ -20,6 +20,8 @@
     IPCChannel,
     LegacyPinnedMemoryResource,
     MemoryResource,
+    VMMAllocatedMemoryResource,
+    VMMAllocationOptions,
```

Comment:
1. These two names should align -- one is "allocated" another is "allocation"
   - @Andy-Jost we forgot to expose `DeviceMemoryResorceOptions` as public API
2. Instead of `VMMAllocatedMemoryResource`, how about a shorter, more informative one: `VirtualMemoryResource`?

Reply:
How's `VirtualMemoryResource`  and `VirtualMemoryResourceOptions` ?